### PR TITLE
Allow metadata to be attached to operations

### PR DESCRIPTION
**Pull request by rubensworks**
I could not find any unit tests related to `mapOperation`, so I couldn't add to them 😨

### CODE REVIEW

File: lib/util.ts [hunk 1]
```diff
@@ -325,117 +325,178 @@ export default class Util
...
+            doRecursion = recurseResult.recurse;
+            copyMetadata = !recurseResult.skipMetadataCopy;
+        }
+
+        const originalMetadata = result.metadata || op.metadata;
+
+        if (!doRecursion) {
+            // Inherit metadata
+            if (originalMetadata && copyMetadata) {
+                result.metadata = originalMetadata;
```
Comment (joachimvh) [inline lib/util.ts:346]: What if `result` also contains metadata. Is it preferred behaviour that this gets removed? If not, perhaps `result.metadata = { ...result.metadata, ...originalMetadata }`. Potentially with another check to not have an empty object if both are undefined.

Comment (rubensworks) [reply lib/util.ts:346]: Sure, we could do that. The user has a way of overriding this behaviour anyways.

**Review by joachimvh — APPROVED** [review]

**Comment by joachimvh** [comment]
Released as v4.3.0

**Comment by rubensworks** [comment]
Thanks 🙂

**Pull request merged**